include/RAJA/index/RangeSegment.hpp

```diff
@@ -102,28 +102,16 @@ struct TypedRangeSegment {
   RAJA_HOST_DEVICE TypedRangeSegment() = delete;
 
   //! move constructor
-  RAJA_HOST_DEVICE constexpr TypedRangeSegment(TypedRangeSegment&& o)
-      : m_begin(std::move(o.m_begin)), m_end(std::move(o.m_end))
-  {
-  }
+  constexpr TypedRangeSegment(TypedRangeSegment&& o) = default;
```

Comment:
@trws for defaulted methods, should we remove argument names to avoid compiler warnings?

Reply:
There shouldn't be any warnings from them, since there's no body, but if you prefer it stylistically I have no issue with removing the names, they don't really add anything at the moment.

Reply:
I'm fine either way. I wasn't sure if there were potential warnings lurking with named args for some compilers. Actually, we have named args in some places and others without. I'll go through and remove the names so everything is consistent.  Are you good with that?